data/src/main/java/com/example/data/mappers/WeatherDataEntityMapper.kt

```diff
@@ -1,32 +1,68 @@
+import com.example.data.db.tables.WeatherDataEntity
+import com.example.data.modelsApi.currentWeather.CurrentWeatherApiModel
+import com.example.data.modelsApi.multiDaysWeather.DailyWeatherApi
 import com.example.data.utils.ICON_URL
+import com.example.data.utils.getCurrentDateByTimezone
+import com.example.data.utils.parsingDate
 import com.example.domain.models.WeatherData
-import java.text.SimpleDateFormat
-import java.util.*
+import kotlin.math.roundToInt
 
 class WeatherDataEntityMapper {
-    @SuppressLint("SimpleDateFormat")
-    fun fromApiToEntity(weatherDataApi: WeatherDataApiModel): WeatherDataEntity {
-        val currentDate = SimpleDateFormat("yyyy-MM-dd").format(Date())
+    fun fromApiToEntity(currentWeatherApi: CurrentWeatherApiModel): WeatherDataEntity {
 
         return WeatherDataEntity(
-            weatherDataApi.city_name,
-            currentDate,
-            weatherDataApi.data[0].temp.toString(),
-            "$ICON_URL${
-            weatherDataApi.data[0].weather.icon}.png"
+            cityName = currentWeatherApi.data[0].city_name,
+            date = getCurrentDateByTimezone(currentWeatherApi.data[0].timezone),
+            temperature = currentWeatherApi.data[0].temp.roundToInt().toString(),
+            icon = "$ICON_URL${
+            currentWeatherApi.data[0].weather.icon}.png",
+            description = currentWeatherApi.data[0].weather.description
         )
     }
 
+    fun fromApiToEntityList(dailyWeatherApi: DailyWeatherApi): List<WeatherDataEntity> {
+        val list = mutableListOf<WeatherDataEntity>()
+
+        for (i in 1 until  dailyWeatherApi.data.size) { //The loop starts from the second element of the list because the first element (current day) is not included
+            list.add(
+                WeatherDataEntity(
+                    cityName = dailyWeatherApi.city_name,
+                    date = dailyWeatherApi.data[i].datetime,
+                    temperature = dailyWeatherApi.data[i].max_temp.roundToInt().toString(),
+                    icon = "$ICON_URL${dailyWeatherApi.data[i].weather.icon}.png",
+                    description = dailyWeatherApi.data[i].weather.description
+                )
+            )
+        }
+        return list
+    }
+
     fun fromEntityToWeatherData(weatherDataEntity: WeatherDataEntity): WeatherData {
         return WeatherData(
-            weatherDataEntity.cityName,
-            weatherDataEntity.temperature,
-            weatherDataEntity.icon
+            city_name = weatherDataEntity.cityName,
+            temp = weatherDataEntity.temperature,
+            icon = weatherDataEntity.icon,
+            date = parsingDate(weatherDataEntity.date),
+            description = weatherDataEntity.description
         )
     }
-}
\ No newline at end of file
+
+    fun fromEntityListToWeatherDataList(entityList: List<WeatherDataEntity>): List<WeatherData> {
+        val list = mutableListOf<WeatherData>()
+
+        for (element in entityList) {
```

Comment:
Could be implemented without for loop and list.add: 
list = entityList.map { ... }

Reply:
Fixed

Reply:
You probably can remove list at all:
`return entityList.map { ... }`